actions/shared-node-cache/action.yml

```diff
@@ -5,26 +5,40 @@ inputs:
     description: 'Node version to use'
     required: false
     default: '20.x'
+  cache-cypress:
+    description: 'If true, caches and restore Cypress packages'
+    required: false
+    default: 'false'
+
 runs:
   using: "composite"
   steps:
-    - name: Use Node.js ${{ inputs.node-version }}
-      uses: actions/setup-node@v4
+    - uses: pnpm/action-setup@v4
       with:
-        node-version: ${{ inputs.node-version }}
+        run_install: false
 
-    - name: Cache node_modules and cypress if it exists
-      uses: actions/cache@v4
-      id: cache-node-modules
+    # MUST be before we install node_modules as that depends on finding
+    # Cypress binaries in this cache!
+    - uses: actions/cache@v4
+      if: inputs.cache-cypress == 'true'
       with:
-        path: |
-          node_modules
-          ~/.cache/Cypress
-        key: ${{ runner.os }}-${{ inputs.node-version }}-v1-${{ hashFiles('**/yarn.lock') }}
+        path: ~/.cache/Cypress
+        # Generate a new cache whenever the lock file changes
+        key: ${{ runner.os }}-cypress-${{ hashFiles('**/pnpm-lock.yaml') }}
+        # If source files changed, rebuild from a prior cache.
         restore-keys: |
-          ${{ runner.os }}-${{ inputs.node-version }}-v1-
+          ${{ runner.os }}-cypress-
```

Comment:
One thought I had.  This may cause the Cypress cache to grow indefinitely. On my local machine my cache has many versions of Cypress binaries in it. Should we perhaps _not_ use a fallback restore keys and allow the install to fetch the binaries if the exact version isn't available. 🤔